sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ResourceProviderDataTests.cs

```diff
@@ -52,5 +52,36 @@ public void ValideStjIntegration()
             var stjResult = JsonSerializer.Serialize(modelFromStj, stjOptions);
             Assert.AreEqual(WirePayload, stjResult);
         }
+
+        [Test]
+        public void ValidatePrettyPrintWithStj()
+        {
+            var stjOptions = new JsonSerializerOptions
+            {
+                Converters = { new JsonModelConverter() },
+                WriteIndented = true,
+            };
+
+            var modelFromStj = JsonSerializer.Deserialize<ResourceProviderData>(WirePayload, stjOptions);
+            var stjResult = JsonSerializer.Serialize(modelFromStj, stjOptions);
+            Assert.AreEqual(File.ReadAllText(TestData.GetLocation("ResourceProviderData/ResourceProviderData-TwoSpaces.json")).TrimEnd(), stjResult);
+        }
+
+        [Test]
+        public void ValidateCapitalizationIsIgnored()
+        {
+#if NET8_0_OR_GREATER
+            var stjOptions = new JsonSerializerOptions
+            {
+                Converters = { new JsonModelConverter() },
+                WriteIndented = true,
+                PropertyNamingPolicy = JsonNamingPolicy.KebabCaseUpper,
+            };
+
+            var modelFromStj = JsonSerializer.Deserialize<ResourceProviderData>(WirePayload, stjOptions);
+            var stjResult = JsonSerializer.Serialize(modelFromStj, stjOptions);
```

Comment:
Curious whether this should throw to indicate options were invalid instead of silently fail?  Do we know what STJ behavior would be in such a case?

Reply:
Imagine a case where you have a type X that has a property of type Y and X is not MRW but Y is.  All of the options are "valid" as they would apply to X, but MRW will own the shape of Y since its a custom converter.